File: src/main/kotlin/com/example/demo/entities/Car_.java
package com.example.demo.entities;

import java.time.LocalDateTime;
import javax.annotation.Generated;
import javax.persistence.metamodel.ListAttribute;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(Car.class)
public abstract class Car_ {

	public static volatile SingularAttribute<Car, BodyType> bodyType;
	public static volatile SingularAttribute<Car, Integer> createUserId;
	public static volatile SingularAttribute<Car, Integer> updateUserId;
	public static volatile SingularAttribute<Car, Long> price;
	public static volatile SingularAttribute<Car, String> name;
	public static volatile ListAttribute<Car, Part> parts;
	public static volatile SingularAttribute<Car, LocalDateTime> createdDateTime;
	public static volatile SingularAttribute<Car, User> createUser;
	public static volatile SingularAttribute<Car, Integer> id;
	public static volatile SingularAttribute<Car, LocalDateTime> updatedDateTime;

	public static final String BODY_TYPE = "bodyType";
	public static final String CREATE_USER_ID = "createUserId";
	public static final String UPDATE_USER_ID = "updateUserId";
	public static final String PRICE = "price";
	public static final String NAME = "name";
	public static final String PARTS = "parts";
	public static final String CREATED_DATE_TIME = "createdDateTime";
	public static final String CREATE_USER = "createUser";
	public static final String ID = "id";
	public static final String UPDATED_DATE_TIME = "updatedDateTime";

}

File: src/main/kotlin/com/example/demo/entities/Car.kt
package com.example.demo.entities

import org.hibernate.annotations.Fetch
import org.hibernate.annotations.FetchMode
import java.io.Serializable
import java.time.LocalDateTime
import javax.persistence.*

// 車を表現する
@Entity
@Table(name = "cars")
data class Car(
        // 車ID
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Int?,
        // 車の名前
        @Column(length = 100, nullable = false)
        val name: String,
        // 車種
        @Column(length = 10, nullable = false)
        @Enumerated(EnumType.STRING)
        val bodyType: BodyType,
        // 価格
        @Column(nullable = false)
        val price: Long,
        // 作成者ID
        @Column(nullable = false)
        val createUserId: Int,
        // 作成日時
        @Column(nullable = false)
        val createdDateTime: LocalDateTime,
        // 更新者ID
        @Column(nullable = false)
        val updateUserId: Int,
        // 更新日時
        @Column(nullable = false)
        val updatedDateTime: LocalDateTime,
        @ManyToOne(optional = false)
        @JoinColumn(name = "createUserId", referencedColumnName = "id", insertable = false, updatable = false)
        val createUser: User,
        @OneToMany(mappedBy = "car", cascade = [CascadeType.ALL], orphanRemoval = true)
        @Fetch(FetchMode.JOIN)
        var parts: MutableList<Part>
) : Serializable

// 車種を表現する
enum class BodyType(val view: String, val taxRate: Double) {
    Kei("軽自動車", 1.08),
    Sedan("セダン", 1.1),
    Suv("SUV", 1.1),
    Minivan("ミニバン", 1.1),
    Wagon("ワゴン", 1.1);

    // 税込み価格を取得する
    fun getTaxPrice(price: Long): Long {
        val taxPrice = price * taxRate
        return taxPrice.toLong()
    }
}